components/NavBar.tsx

```diff
@@ -1,50 +1,88 @@
 import Link from 'next/link';
-import React from 'react'
+import React, { useState } from 'react';
+import Image from 'next/image';
 
 interface ActiveOption {
     activeOption?: "About" | "Events" | "Gallery" | "Branches" | "Contact Us" | "CodeRED" | "Register";
 }
+
 const NavBar = () => {
+    const [isBranchesOpen, setBranchesOpen] = useState(false);
+
+    const toggleBranchesDropdown = () => {
+        setBranchesOpen(!isBranchesOpen);
+    };
+
+    const handleMouseEnter = () => {
+        setBranchesOpen(true);
+    };
+
+    const handleMouseLeave = () => {
+        setBranchesOpen(false);
+    };
 
     return (
-      <nav className="flex flex-row justify-between mx-8 mt-2 text-center">
-        <div className="flex justify-between gap-4">
-            <Link href="/">
-                <img
-                src="/icons/logo.svg"
-                alt="Logo"
-                className="mx-auto"
-                />
-            </Link>
-            <h1 className="my-1 font-extrabold text-2xl text-red-700">CougarCS</h1>
-        </div>
-        <ul className="flex gap-4 px-8 mt-4">
-          <li>
-            <Link href="/">About</Link>
-          </li>
-          <li>
-            <Link href="/events">Events</Link>
-          </li>
-          <li>
-            <Link href="/gallery">Gallery</Link>
-          </li>
-          <li>
-            <Link href="/branches">Branches</Link>
-          </li>
-          <li>
-            <Link href="/contactus">Contact Us</Link>
-          </li>
-          <li>
-            <Link href="/codered">CodeRED</Link>
-          </li>
-            <li className="">
-                <Link className="bg-red-700 text-white px-4 py-2" href="/join">Register</Link>
-            </li>
-        </ul>
-      </nav>
+        <nav className="flex flex-row justify-between mx-8 mt-2 text-center">
+            <div className="flex justify-between gap-4">
+                <Link href="/">
+                    <img
+                        src="/icons/logo.svg"
+                        alt="Logo"
+                        className="mx-auto"
+                    />
+                </Link>
+                <h1 className="my-1 font-extrabold text-2xl text-red-700">CougarCS</h1>
+            </div>
+            <ul className="flex gap-4 px-8 mt-4">
+                <li>
+                    <Link href="/">About</Link>
+                </li>
+                <li>
+                    <Link href="/events">Events</Link>
+                </li>
+                <li>
+                    <Link href="/gallery">Gallery</Link>
+                </li>
+                <li
+                    className="relative flex items-center"
```

Comment:
delete flex and items-center from this line, what does this do?

With flex and items-center, "branches" wont be in align with the rest of the navbar. How can we make branches appear on the right without messing up how branches appear?